diff --git a/node_modules/payload/dist/bin/generateTypes.js b/node_modules/payload/dist/bin/generateTypes.js
index 357e56d..f805a31 100644
--- a/node_modules/payload/dist/bin/generateTypes.js
+++ b/node_modules/payload/dist/bin/generateTypes.js
@@ -393,6 +393,7 @@ function generateTypes() {
     }).then((compiled) => {
         fs_1.default.writeFileSync(outputFile, compiled);
         logger.info(`Types written to ${outputFile}`);
+        process.exit(0);
     });
 }
 exports.generateTypes = generateTypes;
