diff --git a/node_modules/serverless-webpack/lib/packExternalModules.js b/node_modules/serverless-webpack/lib/packExternalModules.js
index 9090d4a..2710718 100644
--- a/node_modules/serverless-webpack/lib/packExternalModules.js
+++ b/node_modules/serverless-webpack/lib/packExternalModules.js
@@ -79,9 +79,21 @@ function getProdModules(externalModules, packagePath, nodeModulesRelativeDir, de
     return [];
   }
 
+  // Check whether there a local package.json file is provided
+  let localPackageJson;
+  const includes = this.configuration.includeModules;
+  const localPackageJsonPath = _.get(includes, 'localPackageJson');
+  if (localPackageJsonPath) {
+    localPackageJson = require(path.join(process.cwd(), localPackageJsonPath));
+  }
+
   // Get versions of all transient modules
   _.forEach(externalModules, module => {
     let moduleVersion = packageJson.dependencies[module.external];
+    // Override module versions with the ones provided by local package.json
+    if (localPackageJson && localPackageJson.dependencies[module.external]) {
+      moduleVersion = localPackageJson.dependencies[module.external];
+    }
 
     if (moduleVersion) {
       prodModules.push(`${module.external}@${moduleVersion}`);
